hack/validation/kubelet.sh

```diff
@@ -1,22 +1,25 @@
 # Kubelet Validation 
 
-# The regular expression adds validation for kubelet.kubeReserved and kubelet.systemReserved values of the map are resource.Quantity
-# Quantity: https://github.com/kubernetes/apimachinery/blob/d82afe1e363acae0e8c0953b1bc230d65fdb50e2/pkg/api/resource/quantity.go#L100
-# NodeClaim Validation:
-yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.kubelet.properties.kubeReserved.additionalProperties.pattern = "^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$"' -i pkg/apis/crds/karpenter.sh_nodeclaims.yaml
-yq eval '.spec.versions[0].schema.openAPIV3Schema.properties.spec.properties.kubelet.properties.systemReserved.additionalProperties.pattern = "^(\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))(([KMGTPE]i)|[numkMGTPE]|([eE](\+|-)?(([0-9]+(\.[0-9]*)?)|(\.[0-9]+))))?$"' -i pkg/apis/crds/karpenter.sh_nodeclaims.yaml 
+# Adding validation to both v1 and v1beta1 APIs
```

Comment:
nit: Is there a way to fix this diff so that it only shows the additive changes? Not a huge deal; I'm suspecting that this is just a small set of changes to account for the different versions, but I'm surprised the diff is so large for a few small characters of change

Reply:
A lot of the change is due to spacing adjustment. Since there is a loop included, we see that most of the spacing is adjusted. I can remove the loop and switch to an args based approach of passing the version to add the validation, which I suspect with reduce the number of diff